src/main/java/com/craftingguide/exporter/extensions/minecraft/ItemGatherer.java

```diff
@@ -2,30 +2,73 @@
-import java.util.Set;
+
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
+import net.minecraft.util.IIcon;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+import codechicken.nei.guihook.GuiContainerManager;
+import cpw.mods.fml.common.ModContainer;
+import cpw.mods.fml.common.registry.GameData;
 
 public class ItemGatherer extends Gatherer {
 
     @Override
     public void gather(ModPackModel modPack) {
-        Set<String> ids = (Set<String>) Item.itemRegistry.getKeys();
+    	Set<String> ids = (Set<String>) Item.itemRegistry.getKeys();
         for (String id : ids) {
             Item item = (Item) Item.itemRegistry.getObject(id);
 
+            ModContainer rawMod = GameData.findModOwner(GameData.getItemRegistry().getNameForObject(item));
+            ModModel mod = rawMod == null ? modPack.getMod("minecraft") : modPack.getMod(rawMod.getModId());
+            //TODO: Buildcraft items are gotten 2 times
+            boolean needToAddItemToModModel = true;
+            for (ItemModel itemModel : mod.getAllItems()) {
+            	if (itemModel.getRawItemStack().getItem() == item) {
+            		needToAddItemToModModel = false;
+            	}
+            }
             if (item.getHasSubtypes()) {
                 ArrayList<ItemStack> subItemStacks = new ArrayList<ItemStack>();
                 item.getSubItems(item, null, subItemStacks);
+                
+                if (subItemStacks.isEmpty()) {
+                	// The following has been borrowed from ChickenBones' NEI code at http://bit.ly/2fNyMLI
+                    // BEGIN NEI
+                	
+                	HashSet<String> damageIconSet = new HashSet<String>();
+                    for (int damage = 0; damage < 16; damage++) {
+                    	ItemStack itemstack = new ItemStack(item, 1, damage);
+                        IIcon icon = item.getIconIndex(itemstack);
+                        String name = GuiContainerManager.concatenatedDisplayName(itemstack, false);
+                        String s = name + "@" + (icon == null ? 0 : icon.hashCode());
+                        if (!damageIconSet.contains(s)) {
+                        	damageIconSet.add(s);
+                            subItemStacks.add(itemstack);
+                        }
+                    }
+                    
+                    // END NEI
+                }
 
                 for (ItemStack stack : subItemStacks) {
                     String subTypeId = id + ":" + stack.getItemDamage();
+                    if (needToAddItemToModModel) {
+                    	mod.addItem(new ItemModel(subTypeId, stack));
+                    }
                     modPack.addItem(new ItemModel(subTypeId, stack));
                 }
             } else {
                 ItemStack stack = new ItemStack(item, 1, 0);
+                if (needToAddItemToModModel) {
+                	mod.addItem(new ItemModel(id, stack));
+                }
```

Comment:
It's not clear to me what benefit there is to changing this class.  So far as I can tell, you're not actually gathering a different set of items: you're just doing it in a more complicated fashion.  The exception is that you seem to be gathering damaged variants of items (something I actually do not want on the site).  Am I missing the real purpose here?

Reply:
some mods items weren't gotten like portal guns orange portal gun.

Reply:
Ah, now I see.  No matter what, this would best belong in a new class under the NEI extensions, though.  So, I think you should revert the changes to *this* file, and just add a new `Gatherer` with all this code over there.  It will then be possible to tinker around with disabling this one in the Minecraft extension and enabling the new one under NEI, or to switch back again to compare results.